src/txmempool.cpp

```diff
@@ -187,10 +201,13 @@ bool CTxMemPool::CalculateMemPoolAncestors(const CTxMemPoolEntry &entry, setEntr
         staged_ancestors.erase(stage);
         totalSizeWithAncestors += stageit->GetTxSize();
 
-        if (stageit->GetSizeWithDescendants() + entry.GetTxSize() > limitDescendantSize) {
+        // When multiple transactions are passed in, the ancestors and descendants of all transactions
+        // considered together must be within limits even if they are not interdependent. This may be
+        // stricter than the limits for each individual transaction.
+        if (stageit->GetSizeWithDescendants() + total_virtual_size > limitDescendantSize) {
             errString = strprintf("exceeds descendant size limit for tx %s [limit: %u]", stageit->GetTx().GetHash().ToString(), limitDescendantSize);
             return false;
-        } else if (stageit->GetCountWithDescendants() + 1 > limitDescendantCount) {
+        } else if (stageit->GetCountWithDescendants() + total_count > limitDescendantCount) {
             errString = strprintf("too many descendants for tx %s [limit: %u]", stageit->GetTx().GetHash().ToString(), limitDescendantCount);
```

Comment:
Added a "possibly" to the beginning of the error strings, gated on `total_count > 1`, so that the existing tests that assert expected error messages don't need to be changed.